purchase.py

```diff
@@ -56,27 +56,35 @@ def create_intercompany_sale(self):
 
             default_values = Sale.default_get(Sale._fields.keys(),
                     with_rec_name=False)
-
             party = Party(self.company.party.id)
             sale = Sale(**default_values)
+            if not sale.warehouse and Sale.warehouse.required:
```

Comment:
I understand you meant `sale.warehouse.required` instead of `Sale.warehouse.required`.

Reply:
I not agree because "sale" is an object and "Sale" is a class:

```
>>> sale, = Sale.search([], limit=1)
>>> sale.warehouse.required
Traceback (most recent call last):
  File "/tryton/trytond/trytond/model/modelstorage.py", line 1608, in __getattr__
    field = self._fields[name]
            ~~~~~~~~~~~~^^^^^^
KeyError: 'required'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/tryton/trytond/trytond/model/modelstorage.py", line 1610, in __getattr__
    raise AttributeError('"%s" has no attribute "%s"' % (self, name))
AttributeError: "stock.location,224" has no attribute "required"
```